rpc/connection.go

```diff
@@ -609,6 +609,9 @@ func (c *Connection) doReconnect(ctx context.Context, disconnectStatus Disconnec
 			// A fatal error happened.
 			*reconnectErrPtr = err
 			// short-circuit Retry
+			if err != nil {
+				c.transport.Close()
```

Comment:
In other words, I think we are calling `Dial` regardless on these reconnect tries, he `Close` I stuck in just makes sure the fd from the previous attempt gets cleaned up. I could be missing something though.